### DISCUSSION TITLE
C# code snippet is not working

### DISCUSSION THREAD

**Discussion by muhssamy** [TabularEditor/TabularEditor3#1302] [🙋♀️TE3 Q&A] [answered]
Dear All
i copied this code from documentations and it is not working with me and i don't know why could any one help me on this ?

source :- https://docs.tabulareditor.com/te2/Useful-script-snippets.html

```
// quote start
foreach(var obj in Selected.OfType<ITabularNamedObject>()) {
    var oldName = obj.Name;
    var newName = new System.Text.StringBuilder();
    for(int i = 0; i < oldName.Length; i++) {
        // First letter should always be capitalized:
        if(i == 0) newName.Append(Char.ToUpper(oldName[i]));

        // A sequence of two uppercase letters followed by a lowercase letter should have a space inserted
        // after the first letter:
        else if(i + 2 < oldName.Length && char.IsLower(oldName[i + 2]) && char.IsUpper(oldName[i + 1]) && char.IsUpper(oldName[i]))
        {
            newName.Append(oldName[i]);
            newName.Append(" ");
        }

        // All other sequences of a lowercase letter followed by an uppercase letter, should have a space
        // inserted after the first letter:
        else if(i + 1 < oldName.Length && char.IsLower(oldName[i]) && char.IsUpper(oldName[i+1]))
        {
            newName.Append(oldName[i]);
            newName.Append(" ");
        }
        else
        {
            newName.Append(oldName[i]);
        }
    }
    obj.Name = newName.ToString();
} 
//Quote End
```
sample of my model
![image](https://github.com/TabularEditor/TabularEditor3/assets/49695385/d4cccdd5-d8e2-400e-bca8-9aa2b49991bb)

script running result

![image](https://github.com/TabularEditor/TabularEditor3/assets/49695385/bce6046e-c97b-4d9e-8f35-6fa89b78c1ae)

**Comment by otykier** [comment, marked as answer]
Are you working against a model in Power BI Desktop? If so, please be aware that [Power BI Desktop does not support making changes to tables through external tools](https://learn.microsoft.com/en-us/power-bi/transform-model/desktop-external-tools#data-modeling-operations), so Tabular Editor will not allow changing the name of a table (also when attempting to change the name through a script). You can change this behavior under **Tools > Preferences > Power BI** by enabling the **Allow unsupported modeling operations**, but keep in mind this could cause issues in your .pbix file, so make sure to save a backup copy of the file just in case.

**Reply by muhssamy** [reply]
Thanks For your reply.
